mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java

```diff
@@ -69,22 +82,445 @@
+        private MailboxQuery.UserBound allMailboxesSearchQuery;
+        private MailboxQuery.UserBound inboxSearchQuery;
+        private MailboxQuery.UserBound inboxRenamedSearchQuery;
+
+        @BeforeEach
+        void setUp() {
+            inboxId = CassandraId.timeBased();
+            inboxPath = MailboxPath.forUser(USER, INBOX);
+            inbox = new Mailbox(inboxPath, UID_VALIDITY, inboxId);
+
+            inboxPathRenamed = MailboxPath.forUser(USER, INBOX_RENAMED);
+            inboxRenamed = new Mailbox(inboxPathRenamed, UID_VALIDITY, inboxId);
+            allMailboxesSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPath)
+                .expression(Wildcard.INSTANCE)
+                .build()
+                .asUserBound();
+            inboxSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPath)
+                .expression(new ExactName(INBOX))
+                .build()
+                .asUserBound();
+            inboxRenamedSearchQuery = MailboxQuery.builder()
+                .userAndNamespaceFrom(inboxPathRenamed)
+                .expression(new ExactName(INBOX_RENAMED))
+                .build()
+                .asUserBound();
+        }
+
+        @Test
+        void saveOnCreateShouldBeConsistentWhenFailToPersistMailbox() {
+            doReturn(Mono.error(new RuntimeException("mock exception")))
+                .when(mailboxDAO)
+                .save(inbox);
+
+            doQuietly(() -> testee.save(inbox));
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThatThrownBy(() -> testee.findMailboxById(inboxId))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThatThrownBy(() -> testee.findMailboxByPath(inboxPath))
+                    .isInstanceOf(MailboxNotFoundException.class);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .isEmpty();
+                softly.assertThat(testee.findMailboxWithPathLike(allMailboxesSearchQuery))
+                    .isEmpty();
+            });
+        }
+
+        @Test
+        void saveOnRenameThenFailToGetACLShouldBeConsistentWhenFindByInbox() throws Exception {
+            testee.save(inbox);
+
+            when(aclMapper.getACL(inboxId))
+                .thenReturn(Mono.error(new RuntimeException("mock exception")))
+                .thenCallRealMethod();
+
+            doQuietly(() -> testee.save(inboxRenamed));
+
+            SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                assertMailboxEqualsTo(softly, testee.findMailboxById(inboxId), inbox);
+                assertMailboxEqualsTo(softly, testee.findMailboxByPath(inboxPath), inbox);
+                softly.assertThat(testee.findMailboxWithPathLike(inboxSearchQuery))
+                    .hasOnlyOneElementSatisfying(searchMailbox -> assertMailboxEqualsTo(softly, searchMailbox, inbox));
```

Comment:
Sorry I don't understand this assertion `assertMailboxEqualsTo(softly, searchMailbox, inbox)`

May I suggest:

```
assertThatMailbox(searchMailbox).isEqualTo(inbox).with(softly);
```

Reply:
Can't we find a way to reuse `MailboxAssert`?

Reply:
Well, now I now we have it :)

Reply:
done